openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -575,80 +631,175 @@ def parse(self, sources, input_format=None):
-                root,
-                "'aromaticity_model' (%s) must be one of the supported models: "
-                % (aromaticity_model, topology.ALLOWED_AROMATICITY_MODELS))
-
-        if (self._aromaticity_model is not None) and (self._aromaticity_model
-                                                      != aromaticity_model):
-            self._raise_parsing_exception(
-                root,
-                "'aromaticity_model' (%s) does not match earlier read 'aromaticity_model' (%s)"
-                % (aromaticity_model, self._aromaticity_model))
+
+        # Ensure that SMIRNOFF is a top-level key of the dict
+        if not('SMIRNOFF' in smirnoff_data.keys()):
+            raise ParseError("'SMIRNOFF' must be a top-level key in the SMIRNOFF object model")
+
+        l1_dict = smirnoff_data['SMIRNOFF']
+        # Check that the aromaticity model required by this parameter set is compatible with
+        # others loaded by this ForceField
+        if 'aromaticity_model' in l1_dict.keys():
+            aromaticity_model = l1_dict['aromaticity_model']
+        else:
+            raise ParseError("'aromaticity_model' attribute must be specified in SMIRNOFF tag")
+        self._register_aromaticity_model(aromaticity_model)
+
+        #if aromaticity_model not in topology.ALLOWED_AROMATICITY_MODELS:
+        #    self._raise_parsing_exception(
+        #        root,
+        #        "'aromaticity_model' (%s) must be one of the supported models: "
+        #        % (aromaticity_model, topology.ALLOWED_AROMATICITY_MODELS))
+        #
+        #if (self._aromaticity_model is not None) and (self._aromaticity_model
+        #                                              != aromaticity_model):
+        #    self._raise_parsing_exception(
+        #        root,
+        #        "'aromaticity_model' (%s) does not match earlier read 'aromaticity_model' (%s)"
+        #        % (aromaticity_model, self._aromaticity_model))
+
+
+
+        # Check that the SMIRNOFF version of this data structure is supported by this ForceField implementation
+        if 'version' in l1_dict.keys():
+            version = l1_dict['version']
+        else:
+            raise ParseError("'version' attribute must be specified in SMIRNOFF tag")
+        self._check_smirnoff_version_compatibility(str(version))
+
+
+        # Go through the subsections, delegating each to the proper ParameterHandler
+
+        # Define keys which are expected from the spec, but have no bearing on system energy
+        l1_spec_keys = ['Author','Date', 'version', 'aromaticity_model']
+
+        for parameter_name in l1_dict.keys():
+            # Skip (for now) cosmetic l1 items
+            if parameter_name in l1_spec_keys:
+                continue
+            # Handle cases where a parameter name has no info (eg. ToolkitAM1BCC)
+            if l1_dict[parameter_name] is None:
+                handler = self.get_handler(parameter_name, {})
```

Comment:
Just to make sure I understand, is this call here to make some sort of validation and make sure that `parameter_name` corresponds to a registered handler?

Reply:
Yes, and `ForceField.get_handler(parameter_name, {})` will actually initialize the ParameterHandler if it doesn't exist. So here, just calling `get_handler` on the `ToolkitAM1BCC` tag will make this ForceField assign AM1BCC charges during system creation.